### ISSUE TITLE
Reduce duplication of 1d base state -> 3d MultiFab calculations

### ISSUE THREAD

**Issue by harpolea** [AMReX-Astro/MAESTROeX#99]
Routines offloaded to the GPU now take the 1d base state mapped to a 3d MultiFab. This mapping is often done within the C++ wrapper function of these routines, which is likely to lead to lots of duplication, the same quantities being mapped repeatedly by different functions). It might be more efficient to do the mapping once (e.g. just after the base state is changed) and carrying this around rather than computing it on the fly every time it's needed.

**Comment by ajnonaka** [comment]
One tricky part is that some of the X_cart temporary multifabs are filled with things like "rho0", where rho0 is passed in to a function and could be rho0_old, rho0_new, or some temporarily created rho_nph.

**Comment by harpolea** [comment]
That's true. I wasn't thinking that this should be done for all of the `X_cart`s, but I certainly noticed that there are an awful lot of `w0_cart`s created, which seems rather inefficient

**Comment by ajnonaka** [comment]
agreed about w0_cart.  makes the code much cleaner too, less to pass around if w0_cart is part of the MAESTRO class.

**Comment by ajnonaka** [comment]
I finished the first part of cleaning up w0_cart.  w0_cart is now part of the MAESTRO class.  the next step is to eliminate the calls that fill in w0_cart from w0.  I think the idea is to fill in w0_cart from w0 immediately after make_w0 is called, and only then.

**Comment by ajnonaka** [comment]
ok done with eliminating unnecessary calls to fill w0_cart from w0.  Still other base state arrays to consider.
